lib/gd32/f1x0/flash.c

```diff
@@ -0,0 +1,153 @@
+/** @defgroup flash_file FLASH
+ *
+ * @ingroup GD32F1x0
+ *
+ * @brief <b>libopencm3 GD32F1x0 FLASH</b>
+ *
+ * @version 1.0.0
+ *
+ * @author @htmlonly &copy; @endhtmlonly 2013
+ * Frantisek Burian <BuFran@seznam.cz>
+ *
+ * @date 14 January 2014
+ *
+ * FLASH memory may be used for data storage as well as code, and may be
+ * programmatically modified. Note that for firmware upload the GD32F1x0
+ * provides a built-in bootloader in system memory that can be entered from a
+ * running program.
+ *
+ * FLASH must first be unlocked before programming. In this module a write to
+ * FLASH is a blocking operation until the end-of-operation flag is asserted.
+ *
+ * @note: don't forget to lock it again when all operations are complete.
+ *
+ * LGPL License Terms @ref lgpl_license
+ */
+
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2013 Frantisek Burian <BuFran@seznam.cz>
```

Comment:
so... why does this file need to exist then?  can't it just use the f1 code as is?

Reply:
In fact I just copied f0 code, as f1 have multi-bank.

Could I replace this file with just an #include to f0 code?

Reply:
As I mentioned in the makefile comment avboe, I'd kinda rather see just referring to the stm32 flash file directly, rather than getting into  #including .c files.